Project/scripts/operate.ts

```diff
@@ -1,9 +1,16 @@
 import deploy from "./deployments";
 import getWalletAccount from "./getWalletAccount";
+import "dotenv/config";
 
 async function main() {
-  const signer = await getWalletAccount();
-  const ballotAddress = await deploy(signer, "Allen", "russ", "Kuro", "IKA");
+  const ownerSigner = await getWalletAccount(process.env.PRIVATE_KEY);
```

Comment:
@allen-hsu  
How if someone use mnemonic instead of private key? 
`ethers.Wallet.fromMnemonic(process.env.MNEMONIC)` 

I usually use mnemonic for testing :)

Reply:
according this 

```
async function main(privateKey: string | undefined) {
  const wallet =
    process.env.MNEMONIC && process.env.MNEMONIC.length > 0
      ? ethers.Wallet.fromMnemonic(process.env.MNEMONIC)
      : new ethers.Wallet(privateKey ?? EXPOSED_KEY);
``` 

MNEMONIC > privateKey > EXPOSED_KEY

or we can put the MNEMONIC is param as well, eg: 
```
async function main(
  privateKey: string | undefined,
  mnemonic: string | undefined
) {
  const wallet =
    mnemonic && mnemonic.length > 0
      ? ethers.Wallet.fromMnemonic(mnemonic)
      : new ethers.Wallet(privateKey ?? EXPOSED_KEY);
  console.log(`Using address ${wallet.address}`);
  const provider = ethers.providers.getDefaultProvider("rinkeby");
  const signer = wallet.connect(provider);
  return signer;
}
```

Reply:
When you have MNEMONIC , the condition will choose the MNEMONIC

Reply:
Correct , but it will pull the data `process.env.PRIVATE_KEY` in the .env file , right? 

`await getWalletAccount(process.env.PRIVATE_KEY);` 
If someone doesn't have `PRIVATE_KEY` in his `.env` but have `MNEMONIC=xxxxxxxxxxxx` , then we are calling `await getWalletAccount(NULL);`  , right?    So error will be thrown?

Reply:
no, because you can see this 

```
  privateKey: string | undefined,
  mnemonic: string | undefined
```

can put undefined as well

Reply:
for now, I'm not setting MNEMONIC, also work fine to me

Reply:
if you don't put MNEMONIC to your env file, will go to the private key
if you don't put PRIVATE_KEY to your env file, will go to the use EXPOSED_KEY

Reply:
Yup ... Agree.